extensions/interactions/NumericInput/directives/numeric-input-validation.service.ts

```diff
@@ -105,6 +113,14 @@ export class NumericInputValidationService {
           case 'Equals':
             var x = (<number>rule.inputs.x);
             setLowerAndUpperBounds(range, x, x, true, true);
+            if (x < 0 && customizationArgs.placeholder.value) {
+              warningsList.push({
+                type: AppConstants.WARNING_TYPES.ERROR,
+                message: (
+                  'Rule ' + (j + 1) + ' input ' +
+                  ' should be greater than or equal to zero. ')
+              });
+            }
```

Comment:
Looks like this code is used multiple times. Should we encapsulate it in a function and use it everywhere?

Reply:
done